### PR TITLE
Remove gtest from project

### PR DESCRIPTION
**Pull request by pbjd**
Download on the fly or allow user to specify using gtest=<path> from the
command line.

### CODE REVIEW

File: test/cpp/Makefile [hunk 1]
```diff
@@ -59,12 +63,21 @@ test_simple_aligner: $(GTEST_OBJECTS) $(PBDAGCON_OBJECTS) SimpleAlignerTest.o
...
+	$(get_gtest)
+	unzip -q $(gtest_version).zip
+	$(RM) $(gtest_version).zip
+
 clean:
 	$(RM) $(SRCDIR)/*.o
 	$(RM) *.o
 	$(RM) test_*
-
+	$(RM) -r $(gtest_version)
```
Comment (pb-cdunn) [inline test/cpp/Makefile]: `${gtest_version}` is not defined if `${gtest}` is not defined.

If you combine all the arguments only a single `rm` line, you're less likely to have a problem.

Or maybe you mean `${GTEST_DIR}`. Maybe you should substitute that everywhere (except in the rule to fetch from URL).

**Comment by pbjd** [comment]
Not quite.  gtest_version *is defined* only when gtest is not defined.  I definitely did not mean $(GTEST_DIR), because I would risk removing a user-provide gtest location.  The design here allows the clean target to operate on optional artifacts created by the build (e.g., gtest).  Thought, I suppose it's somewhat pointless to remove, since it's fairly harmless and I'm not removing downloaded boost headers either.

**Comment by pb-cdunn** [comment]
```
$ rm -r
rm: missing operand
Try 'rm --help' for more information.
```

**Comment by pbjd** [comment]
The makefile variable $(RM) maps to `rm -f', which bypasses that error.
________________________________

**Pull request merged**